uui/src/lib/TextField/TextFieldIcon.svelte

```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
+  import Box from '../Box/box.svelte';
+  import { current_component } from 'svelte/internal';
+  import type { TextFieldIconProps } from './TextField.types';
+
+  type T = $$Generic<EventTarget>;
+  type $$Props = TextFieldIconProps<T>;
+</script>
+
+<Box
+  element="i"
+  eventComponent={current_component}
+  {...$$restProps}
+  class={`mdc-text-field__icon mdc-text-field__icon--${$$restProps.slot.split('Icon')[0]} ${
```

Comment:
Need some clarification on this, I think a variant prop should be used over the splitting because the slot is not technically named

Reply:
So i removed the split and and turned the TextFieldIconProps into an interface that expects a variant.